frontend/src/metabase/core/components/Sortable/SortableList.tsx

```diff
@@ -0,0 +1,95 @@
+import { useState, useMemo, useEffect } from "react";
+import type {
+  DragOverEvent,
+  DragStartEvent,
+  Modifier,
+  SensorDescriptor,
+} from "@dnd-kit/core";
+import { DndContext, DragOverlay } from "@dnd-kit/core";
+import { SortableContext, arrayMove } from "@dnd-kit/sortable";
+import { Sortable } from "./Sortable";
+
+interface useSortableListProps<T> {
+  items: T[];
+  getId: (item: T) => string;
+  renderItem: (item: T) => JSX.Element;
+  onSortStart: (event: DragStartEvent) => void;
+  onSortEnd: ({ id, newIndex }: { id: string; newIndex: number }) => void;
+  disableSort?: boolean;
+  sensors?: SensorDescriptor<any>[];
+  modifiers?: Modifier[];
+}
+
+export const SortableList = <T,>({
+  items,
+  getId,
+  renderItem,
+  onSortStart,
+  onSortEnd,
+  disableSort = false,
+  sensors = [],
+  modifiers = [],
+}: useSortableListProps<T>) => {
+  const [itemIds, setItemIds] = useState<any[]>([]);
+  const [activeItem, setActiveItem] = useState<T | null>(null);
+
+  useEffect(() => {
+    setItemIds(items.map(getId));
+  }, [items, getId]);
+
+  const sortableElements = useMemo(
+    () =>
+      itemIds.map(id => {
+        const item = items.find(item => getId(item) === id);
```

Comment:
Should we have `_.indexBy(item, getId)` to avoid these lookups?

Reply:
If it's more efficient then we can